pkg/cmd/version/version.go

```diff
@@ -15,22 +15,96 @@
 package version
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
 
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 // NOTE: use go build -ldflags "-X github.com/tektoncd/cli/pkg/cmd/version.clientVersion=$(git describe)"
-var clientVersion = "dev"
+var clientVersion = devVersion
+
+const devVersion = "dev"
+const latestReleaseURL = "https://api.github.com/repos/tektoncd/cli/releases/latest"
 
 // Command returns version command
 func Command() *cobra.Command {
+	var check bool
+
 	var cmd = &cobra.Command{
 		Use:   "version",
 		Short: "Prints version information",
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			fmt.Fprintf(cmd.OutOrStdout(), "Client version: %s\n", clientVersion)
+
+			if !check || clientVersion == devVersion {
+				return nil
+			}
+
+			return checkRelease(cmd.OutOrStdout())
 		},
 	}
+
+	cmd.Flags().BoolVarP(&check, "check", "c", false, "check if a newer version is available")
 	return cmd
 }
+
+func checkRelease(outStream io.Writer) error {
+	client := http.Client{Timeout: time.Duration(3 * time.Second)}
+
+	res, err := client.Get(latestReleaseURL)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch the latest version")
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return fmt.Errorf("invalid http status %d, error: %s", res.StatusCode, res.Status)
+	}
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read the latest version response body")
+	}
+
+	var response struct {
+		TagName string `json:"tag_name"`
+		HtmlUrl string `json:"html_url"`
+	}
+	if err := json.Unmarshal(body, &response); err != nil {
+		return errors.Wrap(err, "failed to unmarshal the latest version response body")
+	}
+
+	latest, err := parseVersion(response.TagName)
+	if err != nil {
+		return err
+	}
+
+	current, err := parseVersion(clientVersion)
+	if err != nil {
+		return err
+	}
+
+	if current.LT(*latest) {
+		fmt.Fprintf(outStream, "The latest version v%s is avairable. Please check %s.\n", latest, response.HtmlUrl)
```

Comment:
typo: available

Reply:
How about: 
`A newer version (v%s) of  Tekton CLI is available, please check %s` 
else 
 `You are running the latest version (v%s) of Tekton CLI`